### DISCUSSION TITLE
2.8.0

### DISCUSSION THREAD

**Discussion by EveSunMaple** [EveSunMaple/Frosti#50] [Announcements]
### Features

- 为 Navbar 添加了操作：下滑页面收起，上滑放下
- 重做了代码复制按钮，减少客户端代码，具体样式详见演示

### Refactored

- 整体页面调整（之前一直觉得 “贴屏幕贴得太近了” 这种感觉，现在好很多）包括宽度与字号

### Fix

- 修复了若干问题

---

# EN

### Features

- Added an action for the Navbar: collapse when scrolling down, expand when scrolling up
- Redesigned the code copy button, reducing client-side code, with specific styles shown in the demo

### Refactored

- Overall page adjustments (previously, it always felt "too close to the screen," but now it feels much better), including width and font size

### Fix

- Fixed several issues

**Full Changelog**: https://github.com/EveSunMaple/Frosti/compare/v2.8.0-rc...v2.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/EveSunMaple/Frosti/releases/tag/v2.8.0'>2.8.0</a>.</em>

**Comment by JIeJaitt** [comment]
@EveSunMaple  现在的代码块样式看起来没有最开始的那一版舒服放松，现在的偏白色和背景很像，看起来有点费眼睛。怎么会突然更换样式呢

**Reply by EveSunMaple** [reply]
> @EveSunMaple 现在的代码块样式看起来没有最开始的那一版舒服放松，现在的偏白色和背景很像，看起来有点费眼睛。怎么会突然更换样式呢

代码框的样式更换已经过去很久了：

1. 在 [2.3.1](https://github.com/EveSunMaple/Frosti/tree/v2.3.1) 首次添加了白天样式
  ![image](https://github.com/user-attachments/assets/f5d198b1-e39d-48df-ad81-b5390b86645a)
2. 在 [v2.5.0-alpha.5](https://github.com/EveSunMaple/Frosti/tree/v2.5.0-alpha.5) -> [v2.5.0-beta](https://github.com/EveSunMaple/Frosti/tree/v2.5.0-beta) 的更新中使用了转换器自定义了代码框格式
  ![image](https://github.com/user-attachments/assets/bda50714-efc2-4cd6-b6ea-883e8e487e3e)

添加白天样式是为了更好地与当前主题相适应，同样的，在黑夜模式下，代码框也会变成黑夜主题。

如果你需要自己更改代码框的昼夜主题，我已经在配置文件中写好，给出 shiki 提供的主题名称即可。需要注意的是，代码框的工具栏和背景是由 daisyUI 控制的，而代码高亮是由 shiki 控制的。所以，如果你希望在白天显示黑夜主题，可能需要特别修改 global.scss 中的内容
